abdiff/core/run_ab_transforms.py

```diff
@@ -142,6 +175,22 @@ def wait_for_containers(containers: list[Container], timeout: int = 180) -> None
         time.sleep(0.25)
 
 
+def aggregate_logs(run_directory: str, containers: list[Container]) -> str:
+    """Retrieve logs for containers in a list, aggregating to a single log file."""
+    log_file = str(Path(run_directory) / "transformed/logs.txt")
+    with open(log_file, "w") as file:
+        for container in containers:
+            header = (
+                f"docker_image: {container.labels['docker_image']} | "
+                f"source: {container.labels['source']} | "
+                f"input_file: {container.labels['input_file']}\n"
+            )
+            file.write(header)
+            file.write(container.logs().decode())
+            file.write("\n\n")
+    return log_file
```

Comment:
I like that it returns the log filepath; this feels consistent with files as input/ouput for most functionality in this application.